readthedocs/rtd_tests/tests/test_sync_versions.py

```diff
@@ -152,6 +153,498 @@ def test_new_tag_update_inactive(self):
         version_8 = Version.objects.get(slug='0.8.3')
         self.assertFalse(version_8.active)
 
+    def test_normal_behavior_for_stable_after_deleting_user_defined_tag(self):
+        """
+        The user creates a tag named ``stable`` on an existing repo,
+        when syncing the versions, the RTD's ``stable`` is lost
+        and doesn't update automatically anymore, when the tag is deleted
+        on the user repository, the RTD's ``stable`` is back.
+        """
+        version8 = Version.objects.create(
+            project=self.pip,
+            identifier='0.8.3',
+            verbose_name='0.8.3',
+            type=TAG,
+            active=False,
+            machine=False,
+        )
+        self.pip.update_stable_version()
+        current_stable = self.pip.get_stable_version()
+
+        # 0.8.3 is the current stable
+        self.assertEqual(
+            version8.identifier,
+            current_stable.identifier
+        )
+        self.assertTrue(current_stable.machine)
+
+        version_post_data = {
+            'branches': [
+                {
+                    'identifier': 'origin/master',
+                    'verbose_name': 'master',
+                },
+            ],
+            'tags': [
+                # User new stable
+                {
+                    'identifier': '1abc2def3',
+                    'verbose_name': 'stable',
+                },
+                {
+                    'identifier': '0.8.3',
+                    'verbose_name': '0.8.3',
+                },
+            ],
+        }
+
+        resp = self.client.post(
+            '/api/v2/project/{}/sync_versions/'.format(self.pip.pk),
+            data=json.dumps(version_post_data),
+            content_type='application/json',
+        )
+        self.assertEqual(resp.status_code, 200)
+
+        current_stable = self.pip.get_stable_version()
+        version_stable = self.pip.versions.get(slug='stable')
+        self.assertEqual(
+            current_stable.identifier,
+            version_stable.identifier
+        )
```

Comment:
Immediately after this, we want to know is that the `current_stable.machine` is `False` since now it's a version/tag managed by the user.

Reply:
and the `current_stable.identifier` should be `1abc2def3`

Reply:
I believe that is covered by other tests, I'm on the cellphone, so not sure if that is the case

Reply:
The tests from `TestStableVersion` already check that

https://github.com/rtfd/readthedocs.org/blob/486a9f1a06071043f4bb8098a3416f988e293984/readthedocs/rtd_tests/tests/test_sync_versions.py#L1026-L1032